File: Source/AlphaTab/Rendering/Staves/StaveTrackGroup.cs
using AlphaTab.Collections;
using AlphaTab.Model;

namespace AlphaTab.Rendering.Staves
{
    internal class StaveTrackGroup
    {
        public Track Track { get; set; }
        public StaveGroup StaveGroup { get; set; }
        public FastList<Staff> Staves { get; set; }
        public FastList<Staff> StavesRelevantForBoundsLookup { get; set; }

        public Staff FirstStaffInAccolade { get; set; }
        public Staff LastStaffInAccolade { get; set; }

        public StaveTrackGroup(StaveGroup staveGroup, Track track)
        {
            StaveGroup = staveGroup;
            Track = track;
            Staves = new FastList<Staff>();
            StavesRelevantForBoundsLookup = new FastList<Staff>();
        }

        public void AddStaff(Staff staff)
        {
            Staves.Add(staff);
            if (staff.IsRelevantForBoundsLookup)
            {
                StavesRelevantForBoundsLookup.Add(staff);
            }
        }
    }
}

File: src/rendering/staves/StaveTrackGroup.ts
import { Track } from '@src/model/Track';
import { RenderStaff } from '@src/rendering/staves/RenderStaff';
import { StaveGroup } from '@src/rendering/staves/StaveGroup';

export class StaveTrackGroup {
    public track: Track;
    public staveGroup: StaveGroup;
    public staves: RenderStaff[] = [];
    public stavesRelevantForBoundsLookup: RenderStaff[] = [];
    public firstStaffInAccolade: RenderStaff | null = null;
    public lastStaffInAccolade: RenderStaff | null = null;

    public constructor(staveGroup: StaveGroup, track: Track) {
        this.staveGroup = staveGroup;
        this.track = track;
    }

    public addStaff(staff: RenderStaff): void {
        this.staves.push(staff);
        if (staff.isRelevantForBoundsLookup) {
            this.stavesRelevantForBoundsLookup.push(staff);
        }
    }
}
